### PR TITLE
Fix: HttpClient leak in Strict mode

### PR DESCRIPTION
**Pull request by sojanpr**
Preventing an SMV issue with HTTP Post and GET which can delay the
request response.
@aaustin @qinweigong @shortstuffsushi

### CODE REVIEW

**Comment by shortstuffsushi** [comment]
Isn't this shutting down between every single request? If you're calling that on the same instance, does it "restart" on the next request, or will subsequent requests fail due to the connection manager being shut down?

File: Branch-SDK/src/io/branch/referral/RemoteInterface.java [hunk 1]
```diff
@@ -220,6 +222,10 @@ private ServerResponse make_restful_get(String baseUrl, String tag, int timeout,
         } catch (IOException ex) {
             if (log) PrefHelper.Debug(getClass().getSimpleName(), "IO exception: " + ex.getMessage());
             return new ServerResponse(tag, 500);
+        } finally {
+            if (httpClient != null && httpClient.getConnectionManager() != null) {
```
Comment (shortstuffsushi) [inline Branch-SDK/src/io/branch/referral/RemoteInterface.java]: You could encapsulate this inside a separate function:

```
private HttpResponse executeRequest(HttpRequest request) {
     HttpResponse resp = null;
     HttpClient client = null;
     try {
         client = getGenericHttpClient(timeout);
         resp = client.execute(request);
     }
     finally {
         if (client != null && client.getConnectionManager() != null) {
             client.getConnectionManager().shutdown();
         }
     }

     return resp;
}
```

Comment (shortstuffsushi) [inline Branch-SDK/src/io/branch/referral/RemoteInterface.java]: You could do most of this for the two functions, really all you need to do is construct the Request outside and pass it in.

**Comment by sojanpr** [comment]
The `getGenericHttpClient` is creating a new instance of HttpClient for each request
It's not safe to reuse same HttpClient, since it is not thread safe. Alternative approach is  to  use `'AndroidHttpClient` which is thread safe.

   Creating a new method for executing for get and post request looks nice .I really wish to keep the changes minimum for this Hot fix. Will add this change with next  release if thats good.

**Comment by shortstuffsushi** [comment]
Yeah, that's fine. Looks OK to me then :+1:

**Comment by aaustin** [comment]
:+1:

**Pull request merged**